Tests/ImagineEngineTests/Mocks/TextureErrorHandlerMock.swift

```diff
@@ -0,0 +1,25 @@
+//
+//  TextureErrorHandlerMock.swift
+//  ImagineEngine
+//
+//  Created by Vijay on 21/11/17.
+//  Copyright © 2017 ImagineEngine. All rights reserved.
+//
+
+import Foundation
+@testable import ImagineEngine
+
+class TextureErrorHandlerMock: TextureErrorHandler {
+
+    var didIgnore: Bool { return !didLog && !didAssert }
```

Comment:
Think it's better to put this kind of logic inside of the test itself. You usually don't want mocks to contain any logic, but instead just be dumb objects that capture values and method calls.

Reply:
Removed this logic here and instead testing falseness of the other two properties in the `ignore` case in the test. Should be okay right?